presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -382,6 +394,31 @@ private void loadPartition(HivePartitionMetadata partition)
         return list;
     }
 
+    private static List<Path> getTargetPathsFromSymlink(Configuration conf, Path symlinkDir)
+    {
+        try {
+            FileSystem fileSystem = symlinkDir.getFileSystem(conf);
+            FileStatus[] symlinks = fileSystem.listStatus(symlinkDir, HIDDEN_FILES_PATH_FILTER);
+            List<Path> targets = new ArrayList<>();
+
+            for (FileStatus symlink : symlinks) {
+                try (BufferedReader reader = new BufferedReader(new InputStreamReader(fileSystem.open(symlink.getPath())))) {
```

Comment:
I just copied this from the Hadoop code, so this uses the same charset it previously did